okhttp/src/main/java/okhttp3/internal/connection/ExchangeFinder.kt

```diff
@@ -287,12 +291,21 @@ class ExchangeFinder(
       if (nextRouteToTry != null) {
         return true
       }
+
       if (retryCurrentRoute()) {
         // Lock in the route because retryCurrentRoute() is racy and we don't want to call it twice.
         nextRouteToTry = call.connection!!.route()
         return true
       }
-      return (routeSelection?.hasNext() ?: false) || routeSelector.hasNext()
+
+      // If we have a routes left, use 'em.
+      if (routeSelection?.hasNext() == true) return true
+
+      // If we haven't initialized the route selector yet, assume it'll have at least one route.
+      val localRouteSelector = routeSelector ?: return true
```

Comment:
Are these the lines that defer proxy lookup?

Reply:
You got it.

Reply:
Nope, I read this wrong. Deferring the proxy lookup happens by deferring RouteSelector construction.